internal/core/src/segcore/Collection.cpp

```diff
@@ -47,4 +47,20 @@ Collection::parse() {
     schema_ = Schema::ParseFrom(collection_schema);
 }
 
+void
+Collection::parseIndexMeta(const std::string_view index_meta_proto_) {
+    Assert(!index_meta_proto_.empty());
+
+    milvus::proto::segcore::CollectionIndexMeta protobuf_indexMeta;
+    auto suc = google::protobuf::TextFormat::ParseFromString(
+        std::string(index_meta_proto_), &protobuf_indexMeta);
+
+    if (!suc) {
+        std::cerr << "unmarshal index meta string failed" << std::endl;
```

Comment:
use LOG other than stderr

Reply:
Should we directly return here instead of continuing execution?

Reply:
Yes, I have optimized this part, and just return if parse failed.